src/devices/MemoryLCD/LSxxxB7DHxx.cs

```diff
@@ -0,0 +1,300 @@
+
+            buffer[0] = (byte)(ModeSelectionPeriodByte.Mode | (frameInversion ? ModeSelectionPeriodByte.FrameInversion : ModeSelectionPeriodByte.Dummy));
+
+            for (int i = 0; i < lineIndex.Length; i++)
+            {
+                int from = BytesPerLine * i;
+                int to = 1 + (2 + BytesPerLine) * i;
+
+                buffer[to] = Utility.GetAgByte(lineIndex[i] + 1);
+
+                Span<byte> fromLine = data.Slice(from, BytesPerLine);
+                Span<byte> toLine = buffer.AsSpan(to + 1, BytesPerLine);
+
+                fromLine.CopyTo(toLine);
+            }
+
+            WriteSpi(buffer);
+        }
+
+        /// <summary>
+        /// Maintains memory internal data (maintains current display).
+        /// </summary>
+        public void Display(bool frameInversion = false)
+        {
+            // m(1), dummy(1)
+            byte[] buffer = new byte[2];
+
+            buffer[0] = (byte)(frameInversion ? ModeSelectionPeriodByte.FrameInversion : ModeSelectionPeriodByte.Dummy);
+
+            WriteSpi(buffer);
+        }
+
+        /// <summary>
+        /// Clears memory internal data and writes white.
+        /// </summary>
+        public void AllClear(bool frameInversion = false)
+        {
+            // m(1), dummy(1)
+            byte[] buffer = new byte[2];
+
+            buffer[0] = (byte)(ModeSelectionPeriodByte.AllClear | (frameInversion ? ModeSelectionPeriodByte.FrameInversion : ModeSelectionPeriodByte.Dummy));
+
+            WriteSpi(buffer);
+        }
+
+        /// <summary>
+        /// Show image to device
+        /// </summary>
+        /// <param name="image">Image to show</param>
+        /// <param name="split">Number to splits<br/>To avoid buffer overflow exceptions, it needs to split one frame into multiple sends for some device.</param>
+        /// <param name="frameInversion">Frame inversion flag</param>
+        public void ShowImage(Bitmap image, int split = 1, bool frameInversion = false)
+        {
+            FillImageBufferWithImage(image);
+
+            int linesToSend = PixelHeight / split;
+            int bytesToSend = _frameBuffer.Length / split;
+
+            for (int fs = 0; fs < split; fs++)
+            {
+                Span<byte> lineNumbers = _lineNumberBuffer.AsSpan(linesToSend * fs, linesToSend);
+                Span<byte> bytes = _frameBuffer.AsSpan(bytesToSend * fs, bytesToSend);
+
+                DataUpdateMultipleLines(lineNumbers, bytes, frameInversion);
+            }
+        }
+
+        private void Init()
+        {
+            if (_gpio != null)
+            {
+                if (_scs > -1)
+                {
+                    _gpio.OpenPin(_scs, PinMode.Output);
+                }
+
+                if (_disp > -1)
+                {
+                    _gpio.OpenPin(_disp, PinMode.Output);
+                    _gpio.Write(_disp, PinValue.High);
+                    Thread.Sleep(s_powerup_disp_delay);
```

Comment:
See comment at the beginning of this file about using DelayHelper. See here: https://github.com/dotnet/iot/blob/master/src/devices/Common/System/Device/DelayHelper.cs